common/http_error_response_writer.go

```diff
@@ -10,16 +10,52 @@ type HTTPError struct {
 	HTTPCode    int    `json:"-"`
 	Code        string `json:"code,omitempty"`
 	Description string `json:"description,omitempty"`
+
+	extraFields map[string]interface{}
+}
+
+func (httpError *HTTPError) AddField(key string, val interface{}) {
+	if httpError.extraFields == nil {
+		httpError.extraFields = map[string]interface{}{}
+	}
+	httpError.extraFields[key] = val
+}
+
+type KV struct {
+	K string
+	V interface{}
+}
+type wrappedError struct {
+	e      error
+	fields []KV
+}
+
+func (w wrappedError) Error() string {
+	return w.e.Error()
+}
+func WrappedError(err error, fields ...KV) error {
+	return wrappedError{
+		e:      err,
+		fields: fields,
+	}
 }
 
 type httpErrorResponse struct {
-	Status *HTTPError `json:"status"`
+	Status interface{} `json:"status"`
 }
 
 func (httpError *HTTPError) WriteError(ctx context.Context, w http.ResponseWriter) {
```

Comment:
Just thinking, what if we made this an interface method rather than an object method, then the additional fields below can come from another interface, then we don't need to touch the HTTPError struct itself?

Reply:
I like this suggestion. Something like:
```go
type WritableHTTPResponse interface {  // new type introduced
    Write(ctx context.Context, w http.ResponseWriter)
}

type RestGenCallback interface {
    MapError(ctx context.Context, err error) *common.WritableHTTPResponse  // method modified to suit
}
```

@gkanz Do you have any thoughts?

Reply:
That would be a breaking change for other users though.